sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -586,3 +600,295 @@ def finalize(self, metrics):
+
+        for bracket in self.brackets:
+            sentences.extend(self.extract_text_within_brackets(text_in, bracket))
+
+        for sentence in sentences:
+            new_entry = data_entry.copy()
+            new_entry[self.text_key] = sentence
+            # new_entry["ORIGINAL TEXT"] = text_in  # for testing
+            data.append(new_entry)
+
+        data_list = []
+        for data_point in data:
+            data_list.append(DataEntry(data=data_point))
+
+        return data_list
+
+
+class GetWER(BaseParallelProcessor):
+    def __init__(
+        self,
+        text_key: str = "text",
+        pred_text_key: str = "pred_text",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.text_key = text_key
+        self.pred_text_key = pred_text_key
+
+    def process_dataset_entry(self, data_entry) -> List:
+        data_entry['wer'] = get_wer(data_entry[self.text_key], data_entry[self.pred_text_key])
+        return [DataEntry(data=data_entry)]
+
+
+class MakeSentence(BaseParallelProcessor):
+    """
+    Takes string from text string, makes first symbol uppercae and adds end_symbol if last symbol is alpha.
+    """
+
+    def __init__(
+        self,
+        text_key: str = "text",
+        end_symbol: str = ":",
+        make_uppercase: bool = True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.make_uppercase = make_uppercase
+        self.text_key = text_key
+        self.end_symbol = end_symbol
+
+    def process_dataset_entry(self, data_entry) -> List:
+        if self.make_uppercase:
+            data_entry[self.text_key] = data_entry[self.text_key][0].upper() + data_entry[self.text_key][1:]
+
+        # Check if the last character is not a punctuation and if so, add the end_symbol
+        if data_entry[self.text_key][-1].isalpha():
+            data_entry[self.text_key] += self.end_symbol
+        return [DataEntry(data=data_entry)]
+
+
+class ASRFileCheck(BaseProcessor):
+    def __init__(self, audio_filepath_key: str = "audio_filepath", corrupted_audio_dir: str = None, **kwargs):
+        super().__init__(**kwargs)
+        self.audio_filepath_key = audio_filepath_key
+        self.corrupted_audio_dir = corrupted_audio_dir
+        self.failed_files = []
+
+    def process(self):
+        """Check each file listed in the manifest to ensure it can be loaded with torchaudio."""
+        with open(self.input_manifest_file, 'r') as f:
+            lines = f.readlines()
+
+        entries = []
+        total_lines = len(lines)
+
+        for idx in tqdm(range(total_lines), desc="Checking Audio Files"):
+            line = lines[idx]
+            entry = json.loads(line)
+            try:
+                # Attempt to load the audio file to check if it is corrupted
+                torchaudio.load(entry[self.audio_filepath_key])
+                entries.append(entry)  # File is good, append to entries list
+            except Exception as e:
```

Comment:
What kind of error do corrupted files raise? Can we make it more specific than just a general "Exception"?